mongorestore/mongorestore.go

```diff
@@ -198,6 +179,20 @@ func (restore *MongoRestore) ParseAndValidateOptions() error {
 		log.Logv(log.DebugLow, "restoring to a sharded system")
 	}
 
+	restore.isAtlasProxy, err = restore.SessionProvider.IsAtlasProxy()
+	if err != nil {
+		return err
+	}
+	if restore.isAtlasProxy {
+		if restore.InputOptions.RestoreDBUsersAndRoles || restore.ToolOptions.Namespace.DB == "admin" {
+			return fmt.Errorf("cannot restore to the admin database when connected to an atlas proxy")
```

Comment:
"atlas proxy" isn't a term users know. We should refer to them as "MongoDB Atlas free or shared clusters".